apps/shell/app.js

```diff
@@ -78,8 +78,13 @@ default_sshhost = process.env.DEFAULT_SSHHOST || default_sshhost;
 if (default_sshhost) host_whitelist.add(default_sshhost);
 function host_and_dir_from_url(url){
   let match = url.match(host_path_rx),
-  hostname = match[1] === "default" ? default_sshhost : match[1],
-  directory = match[2] ? decodeURIComponent(match[2]) : null;
+  hostname = default_sshhost, 
```

Comment:
So my question below about being broad is really the question of whether in the case that the host regex does not match, do we default the hostname to default_sshhost instead of null? I think if we do the suggestions in the other comments, we can default this to null instead. That way, an incorrectly formed URL errors out gracefully, instead of logging you into the default host.

Reply:
Submitting `https://ondemand-test.osc.edu/pun/dev/shell/this/url/has/no/host` returns `Cannot GET /pun/dev/shell/this/url/has/no/host`. 
There is no other error reporting that might guide the user to the solution though. Another edge case as the user would have to try to break the url.